tests/unit-tests/EmbeddedManagerTests.swift

```diff
@@ -0,0 +1,263 @@
+        var manager: IterableEmbeddedManager? = IterableEmbeddedManager(apiClient: MockApiClient())
+        manager?.onDeinit = {
+            deinitExpectation.fulfill()
+        }
+        
+        XCTAssertEqual(manager?.getMessages().count, 0)
+        manager = nil
+        waitForExpectations(timeout: 1, handler: nil)
+    }
+
+    //didBecomeActiveNotification
+    func testManagerSyncsOnForeground() {
+        let expectation = XCTestExpectation(description: "onMessagesUpdated called")
+        
+        let mockApiClient = MockApiClient()
+        let manager = IterableEmbeddedManager(apiClient: mockApiClient)
+        
+        let mockDelegate = ViewWithUpdateDelegate(
+            onMessagesUpdatedCallback: {
+                expectation.fulfill()
+            },
+            onEmbeddedMessagingDisabledCallback: nil
+        )
+        
+        manager.addUpdateListener(mockDelegate)
+        mockApiClient.populateMessages([
+            IterableEmbeddedMessage(messageId: "1", placementId: 1),
+            IterableEmbeddedMessage(messageId: "2", placementId: 1),
+        ])
+        
+        NotificationCenter.default.post(name: UIApplication.didBecomeActiveNotification, object: nil)
+        
+        wait(for: [expectation], timeout: 5.0)
+    }
+
+    
+    private class ViewWithUpdateDelegate: UIView, IterableEmbeddedUpdateDelegate {
+        init(onMessagesUpdatedCallback: (() -> Void)?, onEmbeddedMessagingDisabledCallback: (() -> Void)?) {
+            super.init(frame: CGRect(x: 0, y: 0, width: 0, height: 0))
+            
+            self.onMessagesUpdatedCallback = onMessagesUpdatedCallback
+            self.onEmbeddedMessagingDisabledCallback = onEmbeddedMessagingDisabledCallback
+        }
+        
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+        
+        private var onMessagesUpdatedCallback: (() -> Void)?
+        private var onEmbeddedMessagingDisabledCallback: (() -> Void)?
+        
+        func onMessagesUpdated() {
+            onMessagesUpdatedCallback?()
+        }
+        
+        func onEmbeddedMessagingDisabled() {
+            onEmbeddedMessagingDisabledCallback?()
+        }
+    }
+    
+    private class MockApiClient: BlankApiClient {
+        private var newMessages = false
+        private var invalidApiKey = false
+        private var mockMessages: [IterableEmbeddedMessage] = []
+
+        func haveNewEmbeddedMessages() {
+            newMessages = true
+        }
+        
+        func populateMessages(_ messages: [IterableEmbeddedMessage]) {
+            self.mockMessages = messages
+            self.newMessages = true
+        }
+        
+        func setInvalidAPIKey() {
+            invalidApiKey = true
+        }
+        
+        override func getEmbeddedMessages() -> IterableSDK.Pending<IterableSDK.PlacementsPayload, IterableSDK.SendRequestError> {
+            if invalidApiKey {
```

Comment:
A little confused here. Does the actual getMessages not get called in the tests?

Reply:
@Ayyanchira if we didn't override getEmbeddedMessages(), the test would attempt to call the API. As this is only a unit test (and not an endpoint test), we don't want to make actual API calls, as that could introduce flakiness and external dependency.